src/main.ts

```diff
@@ -7,8 +7,10 @@ import {promisify} from 'util';
 import * as stream from 'stream';
 
 import * as sorald from './sorald';
+import * as ranges from './ranges';
 
 import * as git from './git';
+import {ClosedRange} from './ranges';
```

Comment:
I think line 10 will take care of this import.

Reply:
This is a single-item import. With only line 10, I'd have to specify `ranges.ClosedRange` everywhere, as opposed to just `ClosedRange`. I don't know what's most common in TypeScript.

Reply:
After looking around a bit more, there seems to be absolutely no consensus around how to do imports in TypeScript. It doesn't help that there are some 6 or so ways to do them. The only thing I can determine to be universally frowned upon is using `require`.

So, I'll go mostly with `import * as whatever` as that seems to be favored in other GHA implementations. I think we can use named imports as well when it makes sense.

Just as a side note on our discussion, I agree with avoiding `*` imports in Python, but that is very different from this `*` import. In python, `import * from module` imports all public top-level members of `module` (or everything defined in `__all__`) as individual objects, whereas `import * as module ...` in TS imports the module by some name (the closest to `import module` in Python that it seems to be possible to get).

Reply:
It appears that combining star import with named import on a single line doesn't work. There appears to be a way to make it work with [synthetic default imports](https://stackoverflow.com/questions/45669863/how-to-combine-import-all-with-others), but restructuring the imports is starting to feel like a separate PR. Perhaps just defer this for later?

This is where a more structured language like Java is nice: there are standard imports, and there are `static` imports, and that's it :)

Reply:
> whereas import * as module ... in TS imports the module by some name

Yes, TS does not have a "default module name" like Python does.

> Perhaps just defer this for later?

Restructuring would require a lot of thought and changes because there will be multiple export statements added and if you export something as default, you would need to decide which one. So you're right that there will be a lot of changed so you may defer this for later.